src/projections/Tools/CommunicationOverTime/CommTimeWindow.java

```diff
@@ -350,59 +350,113 @@ public void showDialog() {
+				readyReaders.add( new ThreadedFileReader(nextPe, intervalSize, startInterval, endInterval, sentMsgCount, receivedMsgCount, sentByteCount, receivedByteCount, receivedExternalMsgCount, receivedExternalByteCount, receivedExternalNodeMsgCount, receivedExternalNodeByteCount) );
+			}
+	
+			// Determine a component to show the progress bar with
+			Component guiRootForProgressBar = null;
+			if(thisWindow!=null && thisWindow.isVisible()) {
+				guiRootForProgressBar = thisWindow;
+			} else if(MainWindow.runObject[myRun].guiRoot!=null && MainWindow.runObject[myRun].guiRoot.isVisible()){
+				guiRootForProgressBar = MainWindow.runObject[myRun].guiRoot;
+			}
+	
+			// Pass this list of threads to a class that manages/runs the threads nicely
+			TimedProgressThreadExecutor threadManager = new TimedProgressThreadExecutor("Loading Communication Data in Parallel", readyReaders, guiRootForProgressBar, true);
+			threadManager.runAll();
 		}
-
-		// Determine a component to show the progress bar with
-		Component guiRootForProgressBar = null;
-		if(thisWindow!=null && thisWindow.isVisible()) {
-			guiRootForProgressBar = thisWindow;
-		} else if(MainWindow.runObject[myRun].guiRoot!=null && MainWindow.runObject[myRun].guiRoot.isVisible()){
-			guiRootForProgressBar = MainWindow.runObject[myRun].guiRoot;
+		else{
+			double[][] sentMsgCount_temp = MainWindow.runObject[myRun].getMsg_count();
+			double[][] sentByteCount_temp = MainWindow.runObject[myRun].getMsg_size();
+			double[][] receivedMsgCount_temp = MainWindow.runObject[myRun].getMsg_recv_count();
+			double[][] receivedByteCount_temp = MainWindow.runObject[myRun].getMsg_recv_size();
+			double[][] receivedExternalMsgCount_temp = MainWindow.runObject[myRun].getMsg_recv_count_ext();
+			double[][] receivedExternalByteCount_temp = MainWindow.runObject[myRun].getMsg_recv_size_ext();
+			int numIntervals2=sentMsgCount_temp[0].length/numEPs;
+			for(int i =0;i<sentMsgCount_temp.length;i++){
+				if((sentMsgCount_temp[i].length/numEPs) > numIntervals2){
+					numIntervals2=sentMsgCount_temp[i].length/numEPs;
+				}			
+			}
+			sentMsgCount = new double[numIntervals][numEPs];
+			sentByteCount = new double[numIntervals][numEPs];
+			receivedMsgCount = new double[numIntervals][numEPs];
+			receivedByteCount = new double[numIntervals][numEPs];
+			receivedExternalMsgCount = new double[numIntervals][numEPs];
+			receivedExternalByteCount = new double[numIntervals][numEPs];
+			receivedExternalNodeMsgCount = new double[numIntervals][numEPs];
+			receivedExternalNodeByteCount = new double[numIntervals][numEPs];
+			
+			for(int i =0;i<sentMsgCount_temp.length;i++){
+				for(int j =0;j<sentMsgCount_temp[i].length;j++){
+					int curInt= j/numEPs ;
+					int curEP= j-curInt*numEPs;
+					curInt = (curInt * 1000)/numIntervals2;
```

Comment:
Where is this `1000` coming from? What exactly is this trying to do? Resample the number of intervals in the logs to be 0-1000?